src/Symfony/Component/String/BinaryString.php

```diff
@@ -0,0 +1,440 @@
+        $str->string = str_pad($this->string, $length, $padStr, STR_PAD_RIGHT);
+
+        return $str;
+    }
+
+    public function padStart(int $length, string $padStr = ' '): parent
+    {
+        $str = clone $this;
+        $str->string = str_pad($this->string, $length, $padStr, STR_PAD_LEFT);
+
+        return $str;
+    }
+
+    public function prepend(string ...$prefix): parent
+    {
+        $str = clone $this;
+        $str->string = (1 >= \count($prefix) ? $prefix[0] ?? '' : implode('', $prefix)).$str->string;
+
+        return $str;
+    }
+
+    public function replace(string $from, string $to): parent
+    {
+        $str = clone $this;
+
+        if ('' !== $from) {
+            $str->string = $this->ignoreCase ? str_ireplace($from, $to, $this->string) : str_replace($from, $to, $this->string);
+        }
+
+        return $str;
+    }
+
+    public function replaceMatches(string $fromPattern, $to): parent
+    {
+        if ('g' === $fromPattern[-1]) {
+            $fromPattern = substr($fromPattern, 0, -1);
+        }
+
+        if ($this->ignoreCase) {
+            $fromPattern .= 'i';
+        }
+
+        if (\is_array($to)) {
+            if (!\is_callable($to)) {
+                throw new \TypeError(sprintf('Argument 2 passed to %s::replaceMatches() must be callable, array given.', \get_class($this)));
+            }
+
+            $replace = 'preg_replace_callback';
+        } else {
+            $replace = $to instanceof \Closure ? 'preg_replace_callback' : 'preg_replace';
+        }
+
+        set_error_handler(static function ($t, $m) { throw new InvalidArgumentException($m); });
+
+        try {
+            if (null === $string = $replace($fromPattern, $to, $this->string)) {
+                $lastError = preg_last_error();
+
+                foreach (get_defined_constants(true)['pcre'] as $k => $v) {
+                    if ($lastError === $v && '_ERROR' === substr($k, -6)) {
+                        throw new RuntimeException('Matching failed with '.$k.'.');
+                    }
+                }
+
+                throw new RuntimeException('Matching failed with unknown error code.');
+            }
+        } finally {
+            restore_error_handler();
+        }
+
+        $str = clone $this;
+        $str->string = $string;
+
+        return $str;
+    }
+
+    public function slice(int $start = 0, int $length = null): parent
+    {
+        $str = clone $this;
+        $str->string = (string) substr($this->string, $start, $length ?? \PHP_INT_MAX);
+
+        return $str;
+    }
+
+    public function splice(string $replacement, int $start = 0, int $length = null): parent
+    {
+        $str = clone $this;
+        $str = substr_replace($this->string, $replacement, $start, $length ?? \PHP_INT_MAX);
+
+        return $str;
+    }
+
+    public function split(string $delimiter, int $limit = null): array
+    {
+        if ('' === $delimiter) {
+            return $this->chunk();
```

Comment:
This is wrong for `$limit !== null`.
(same in other sub classes)

Reply:
now replaced by an exception